src/Cdn77/Sniffs/Classes/ClassStructureSniff.php

```diff
@@ -302,4 +316,102 @@ private function getParentClassName(File $file, int $pointer) : string
 
         return ClassHelper::getName($file, $classPointer);
     }
+
+    private function fixInvalidMemberPlacement(File $file, int $pointer) : void
+    {
+        $tokens = $file->getTokens();
+        $endTypes = [T_OPEN_CURLY_BRACKET, T_CLOSE_CURLY_BRACKET, T_SEMICOLON];
+        $previousMemberEndPointer = TokenHelper::findPrevious($file, $endTypes, $pointer - 1);
+        assert($previousMemberEndPointer !== null);
+
+        $startPointer = $this->findMemberLineStartPointer($file, $pointer, $previousMemberEndPointer);
+
+        if ($tokens[$pointer]['code'] === T_FUNCTION && !FunctionHelper::isAbstract($file, $pointer)) {
+            $endPointer = $tokens[$pointer]['scope_closer'];
+        } else {
+            $endPointer = TokenHelper::findNext($file, T_SEMICOLON, $pointer + 1);
+            assert($endPointer !== null);
+        }
+
+        $possibleWhitespaceTypes = [T_COMMENT, T_DOC_COMMENT, T_DOC_COMMENT_WHITESPACE, T_WHITESPACE];
+        $whitespacePointer = $file->findNext($possibleWhitespaceTypes, $endPointer + 1, null, false, "\n");
+        $nextEffectivePointer = TokenHelper::findNextEffective($file, $endPointer + 1);
+        if ($whitespacePointer < $nextEffectivePointer) {
+            $endPointer = $whitespacePointer;
+        }
+
+        if ($tokens[$previousMemberEndPointer]['code'] === T_CLOSE_CURLY_BRACKET) {
+            $previousScopeOpenerPointer = $tokens[$previousMemberEndPointer]['scope_opener'];
+            $prePreviousMemberEndPointer = TokenHelper::findPrevious($file, $endTypes, $previousScopeOpenerPointer - 1);
+        } else {
+            $prePreviousMemberEndPointer = TokenHelper::findPrevious($file, $endTypes, $previousMemberEndPointer - 1);
+        }
+
+        assert($prePreviousMemberEndPointer !== null);
```

Comment:
Will always be non null? Shouldn't we add exception?

Reply:
I wasn't sure with that ... basically all those asserts are the same: if an assert would fail, then the file wouldn't be valid in the first place (syntax errors, lint would fail)... so ... do we really wan't to validate/handle such cases? Or is an assert enough? And how would we handle it?

Reply:
As `assert.exception` is set to `0` by default, I'd rather explicitly throw exception/error so it's more easily trackable. Guess that would be enough for now. [It seems](https://github.com/slevomat/coding-standard/blob/afccda9f9f497aaf4b426e88dd5dc744d68e04d9/SlevomatCodingStandard/Helpers/Annotation/MethodAnnotation.php#L31) to be common way to handle unexpected inputs.